doc/ifa/SONiC_InbandFlowAnalyzer_FeatureSpec_Ver1.1.md

```diff
@@ -0,0 +1,478 @@
+    3) "devicetype"
+    4) "intermediate"
+
+IFA\_COLLECTOR TABLE
+
+    ;Defines IFA collector configuration
+
+    key                 = name                      ; name is collector name and should be unique.
+    ipaddress-type      = "ipv4" / "ipv6"           ; Collector IP address type.
+    ipaddress           = ipv4_prefix / ipv6_prefix ; Collector IP address.
+    port                = 1 * 4DIGIT                ; Collector UDP port number
+
+    Example:
+    127.0.0.1:6379[4]> keys *IFA_COLLECTOR*
+    1) "IFA_COLLECTOR_TABLE|collector1"
+
+    127.0.0.1:6379[4]> HGETALL "IFA_COLLECTOR_TABLE|collector1"
+    1) "ipaddress-type"
+    2) "ipv4"
+    3) "ipaddress"
+    4) 10.20.30.40
+    5) "port"
+    6) 9070
+
+IFA\_FLOW TABLE
+
+    ;Defines an IFA flow
+
+    key                 = name                ; name is flow name and should be unique
+    acl                 = acl-name            ; Parameter to map to device information to be used for the flow.
+    sampling-rate       = 1 * 5DIGIT          ; Optional parameter that indicates the sampling rate for the packets
+                                              ; belonging to the flow that need to be tagged with IFA meta-data.
+                                              ; One packet in every ‘sampling-rate’ packets will be tagged with IFA metadata.
+                                              ; Note: This parameter is applicable for switches configured as the type ingress-node, and will be ignored for the other types. If this field is
+                                              ; not specified, it indicates a 1:1 sampling.
+                                              ; Range is 1 to 10000. A value of 1 indicates that all packets belonging to this flow will be sampled.
+
+
+    Example:
+    127.0.0.1:6379[4]> keys *IFA_FLOW*
+    1) "IFA_FLOW_TABLE|flow1"
+
+    127.0.0.1:6379[4]> HGETALL "IFA_FLOW_TABLE|flow1"
+    1) "acl-name"
+    2) "acl1"
+    3) "sampling-rate"
+    4) 1000
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+N/A
+
+### 3.2.4 ASIC DB
+The ASIC DB is updated by SAI REDIS upon invocation of SAI REDIS APIs by IFAOrch.
+
+### 3.2.5 COUNTER DB
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+A new orchestration agent class, IFAOrch is added to convert the incoming IFA config to ASIC configuration. IFAOrch subscribes to the IFA tables of CONFIG DB and converts the configuration to the SAI TAM API call sequence described in section 3.5.
+
+IFAOrch maintains data pertaining to all the currently configured IFA entities and the associated TAM object bindings. TAM object bindings are re-used wherever possible.
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+
+The SAI TAM API spec defines all TAM APIs supported in SAI. Please refer to SAI-Proposal-TAM2.0-v2.0.docx in [https://github.com/opencomputeproject/SAI/tree/master/doc/TAM](https://github.com/opencomputeproject/SAI/tree/master/doc/TAM) for more details.
+
```

Comment:
Please specify the SAI API sequence. The referred spec has multiple options for things like bind (source and object), marking the telemetry traffic etc. It would be easier to follow if the SONIC spec lists exactly what APIs and parameters it is using for this release.